### ISSUE TITLE
gtestp4c crashes when testing nullptr

### ISSUE THREAD

**Issue by mbudiu-vmw** [p4lang/p4c#480]
The following line crashes (instead of reporting an error) when the argument is nullptr:
```ASSERT_NE(nullptr, pgm);```

**Comment by sethfowler** [comment]
This is actually my mistake; I believe I took Han to do that, forgetting that GTest wants you to do `ASSERT_TRUE(pgm != nullptr)` instead. `ASSERT_EQ(expected, actual)`, `ASSERT_NE(expected, actual)`, etc are preferred because they give nicer assertion failure messages - along the lines of "We expected 'expected', but we actually got 'actual'."

However, for a type T with overloaded `operator<<(T*)`, if the overloaded operator crashes when passed a null pointer, obviously the assertion will crash when trying to print the assertion failure message. `ASSERT_EQ` handles this case specially, so it's safe to use when the expected value is null, but for some reason Google didn't see fit to support this for the other `ASSERT_*` variants. =\
